lib/tools/apk-utils.js

```diff
@@ -107,12 +110,14 @@ apkUtilsMethods.startUri = async function startUri (uri, pkg) {
  * @throws {Error} If there is an error while executing the activity
  */
 apkUtilsMethods.startApp = async function startApp (startAppOptions = {}) {
-  if (!startAppOptions.activity || !startAppOptions.pkg) {
-    throw new Error('activity and pkg are required to start an application');
+  if (!startAppOptions.pkg || !(startAppOptions.activity || startAppOptions.action)) {
+    throw new Error('pkg, and activity or intent action, are required to start an application');
```

Comment:
I'm not quite sure this condition works as expected

Could you please enumerate which combinations are valid?

Reply:
Sure :)
I just pushed a commit that should verify that (https://github.com/appium/appium-adb/pull/475/commits/05380632160944f7238ed91b8279672822529642) too just in case. It should work like the below, but LMK if you think this isn't right (either code or way it's set up) or if you think we could reword the error message to be clearer. (hoping this is easier to fix than the tests-in-CI was!)

valid:
- pkg + activity (no action)
- pkg + action   (no activity)
- pkg + action + activity *

invalid:
- pkg (no activity + no action)
- activity (no pkg + no action)
- action (no pkg + no activity)

* (this will basically ignore the intent action but it's not invalid and adb will allow it so I didn't validate against it)

Reply:
that's ok, thanks for the update